### PR TITLE
added code for both types of random walks

### PR DESCRIPTION
**Pull request by RidhimaKohli**

### CODE REVIEW

**Comment by czgdp1807** [comment]
Hi @RidhimaKohli 
Thanks for the PR. However, it should be closed in reference to https://github.com/codezonediitj/pydatastructs/wiki/Pull-Request-Policy#when-to-close-a-pr

The code doesn't really looks good and will require a lot of effort in reviewing. Please follow the coding of the rest of the module.

**Comment by RidhimaKohli** [comment]
What are the modifications required and where does the code needs improvement?

File: pydatastructs/graphs/algorithms.py [hunk 1]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
...
+
+def random_walk(G,path_length,rand=random.Random()start=None):
+    '''
+    G is adjacency list 
+    path_length: Length : Length of the random walk.
+    start: the start node of the random walk.'''
+    if start:
+        path=[start]
+    else:
+        path=[rand.choice(list(G.vertices()))]
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:561]: What if the edges are weighted?

File: pydatastructs/graphs/algorithms.py [hunk 2]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+
+def random_walk(G,path_length,rand=random.Random()start=None):
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:553]: What does `rand` parameter is used for?

File: pydatastructs/graphs/algorithms.py [hunk 3]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
...
+    '''
+    G is adjacency list
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:555]: What if adjacency matrix representation or implicit graph representation is used? Do we need different implementer functions for some representations of graphs?

File: pydatastructs/graphs/algorithms.py [hunk 4]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+
+def random_walk(G,path_length,rand=random.Random()start=None):
+    '''
+    G is adjacency list 
+    path_length: Length : Length of the random walk.
+    start: the start node of the random walk.'''
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:557]: It isn't following numpy doc string guide?

File: pydatastructs/graphs/algorithms.py [hunk 5]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
...
+        path=[start]
+    else:
+        path=[rand.choice(list(G.vertices()))]
+    while len(path)<path_length:
+        cur=path[-1]
+        if len(G[cur])>0:
+            path.append(rand.choice(G[cur]))
+        else:
+            break
+        return [str(node) for node in path]
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:568]: `node.name`? May be this will not be a consistent API.

File: pydatastructs/graphs/algorithms.py [hunk 6]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
...
+def loop_erased_random_walk(G,path_length,rand=random.Random(),start=None):
+    if start:
+        path=[start]
+    else:
+        path=[rand.choice(list(G.vertices()))]
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:573]: How does `rand.choice` works? Is it uniformly selecting the next vertex?

File: pydatastructs/graphs/algorithms.py [hunk 7]
```diff
@@ -548,3 +548,36 @@ def strongly_connected_components(graph, algorithm):
...
+
+    while len(path)<path_length:
```
Comment (czgdp1807) [inline pydatastructs/graphs/algorithms.py:575]: Spaces between binary operators aren't there in the patch.

**Comment by czgdp1807** [comment]
I have provided some comments. Many more issues are there with patch. Tests are not added. References are not added. It will take a lot of time and effort in reviewing.

P.S. Please do not make PRs from `master` branch as done here.

**Pull request closed without merging**